microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/utils/AnnotationUtils.java

```diff
@@ -225,4 +234,43 @@ public static Expression getAnnotationMemberValueExpression(Annotation annotatio
 		return null;
 	}
 
+	/**
+	 * Retrieve the value and range of an annotation member given a supported list
+	 * of annotation members
+	 *
+	 * @param annotation            the annotation of the retrieved members
+	 * @param annotationSource      the qualified name of the annotation
+	 * @param annotationMemberNames the supported members of the annotation
+	 * @param position              the hover position
+	 * @param typeRoot              the java type root
+	 * @param utils                 the utility to retrieve the member range
+	 * @return a AnnotationMemberInfo object if the member exists, null otherwise
+	 * @throws JavaModelException
+	 */
+	public static AnnotationMemberInfo getAnnotationMemberInRange(IAnnotation annotation, String annotationSource,
+			String[] annotationMemberNames, Position position, ITypeRoot typeRoot, IJDTUtils utils)
+			throws JavaModelException {
+		String annotationMemberValue = null;
+		for (String annotationMemberName : annotationMemberNames) {
+			annotationMemberValue = getAnnotationMemberValue(annotation, annotationMemberName);
+			if (annotationMemberValue != null) {
+				ISourceRange r = ((ISourceReference) annotation).getSourceRange();

```

Comment:
Add a comment which explain that we retrieve the location of the member value with regexp.